phpBB/posting.php

```diff
@@ -1586,7 +1586,11 @@
 {
 	if ($config['allow_bbcode'])
 	{
-		$message_parser->message = '[quote=&quot;' . $post_data['quote_username'] . '&quot;]' . censor_text(trim($message_parser->message)) . "[/quote]\n";
+		$message_parser->message = $phpbb_container->get('text_formatter.utils')->generate_quote(
+			censor_text(trim($message_parser->message)),
+			array('author' => $post_data['quote_username'])
```

Comment:
Yes. You can put as much metadata as you want, and then use it in the template. I'm wary of overloading the UI though. There's [a topic about possible quote improvements](http://imgur.com/a/Ds9Y0) on A51.